timemachine/fe/rbfe.py

```diff
@@ -408,12 +412,10 @@ def estimate_free_energy_given_initial_states(initial_states, protocol, temperat
 
             # loop over bond, angle, torsion, nonbonded terms etc.
             for u_idx, (prev_U_fn, cur_U_fn) in enumerate(zip(prev_batch_U_fns, cur_batch_U_fns)):
-                x_cur = (cur_frames, cur_boxes)
```

Comment:
`x_cur` and `x_prev` stored references to frames that wouldn't be cleared until the next loop and could hold onto the previous frames even after having sampled the new current frames.

Reply:
Good catch! I think `del x_cur, x_prev` at the end of the loop body would also solve this?